### ISSUE TITLE
Allow the position where an edge should connect to an anchor

### ISSUE THREAD

**Issue by rohanrajpal** [open-source-labs/Svelvet#314]
Currently, if I make an anchor invisible, this is how things will look

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/7023147/235420986-f7a4349d-2f95-4659-a6f6-6340a72d2680.png">

Ideally, I'd want to be able to configure where the edge should connect to an anchor, I want to configure it to something like below

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/7023147/235421276-7f69595e-7f10-4bf0-bf3f-dc47671f97c6.png">

A simple solution is to just mark if the end of edge should be top left or center of an anchor, but in that case the edge will point on absolutely the top left of a node, ideally i want it to be a few pixels below the top left, like above.

**Comment by rohanrajpal** [comment]
Imo a better api than specifying where the edge should end would be to connect the edge to a `CustomAnchor`, if the anchor has one.

So ultimately, the usage would be like

```svelte
<Anchor
        id="anchor-prev"
        let:linked
        let:connecting
        let:hovering
        invisible
        input
        edge={CustomEdge}
    >
        <CustomAnchor {hovering} {connecting} {linked} />
    </Anchor>
    ```

and then in the custom anchor, i can position it wherever i want it to be (since the invisible anchor is anyway spanning over the whole node & the edge would connect to the X & Y of the `CustomAnchor` & not the `Anchor`

```
Using the same custom anchor position as given in svelvet examples, with a little bit of modification

```svelte
<script lang="ts">
    export let linked: boolean;
    export let connecting: boolean;
    export let hovering: boolean;
</script>

<div class:linked class:hovering class:connecting class="my-anchor -right-5 bottom-1" />

<style>
    .my-anchor {
        width: 12px;
        height: 12px;
        border-radius: 50%;
        border: solid 2px white;
        border-color: inherit;
    }
    .hovering {
        scale: 1.2;
    }
    .linked {
        background-color: rgb(213, 245, 246) !important;
    }
    .connecting {
        background-color: rgb(213, 245, 246);
        animation: pulse 1s infinite;
    }
    @keyframes pulse {
        0% {
            box-shadow: 0 0 0 0 rgb(136, 248, 251);
        }
        70% {
            box-shadow: 0 0 0 10px rgba(255, 0, 0, 0);
        }
        100% {
            box-shadow: 0 0 0 0 rgba(255, 0, 0, 0);
        }
    }
</style>
```

**Comment by briangregoryholmes** [comment]
I've changed the implementation of this idea in the latest version and it should no longer conflict with Tailwind.

The relevant prop on the Anchor is now called `nodeConnect` rather than `invisible`. It works regardless of where the Anchor is placed. If you set `nodeConnect` on multiple Anchors, we'll just assign connections to them in order.

The `invisible` prop now just doesn't render out the default Anchor, which is useful in certain cases.

Closing this issue as it is essentially a duplicate of https://github.com/open-source-labs/Svelvet/issues/313 and is associated with an open discussion.
